### PR TITLE
Add Brother mfcj880dw driver

### PR DESCRIPTION
**Pull request by 6543**
I used pkgs/misc/cups/drivers/mfcj6510dw* as template

this is my first nix package - so please be patient :)

~~PS: I still need to test if it works~~ works

### CODE REVIEW

**Comment by 6543** [comment]
did `nix-shell -p nixpkgs-review --run "nixpkgs-review pr 279265"` and it compiled

now i have to figure out, how to use this on my pc to do an actually test them :) ?

**Comment by 6543** [comment]
Testet ... works since some time :)

Can we get it in now?

**Review by sefidel — CHANGES_REQUESTED** [review]
New packages should be defined in `pkgs/by-name` instead.

> This category hierarchy is partially deprecated and will be migrated away over time. The new pkgs/by-name directory ([docs](https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md)) should be preferred instead.

https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#category-hierarchy

**Comment by 6543** [comment]
@sefidel thanks for looking into it. I hope I addressed it correctly :)

**Comment by 6543** [comment]
@Artturin sorry for the ping, i just hope to get some response of how i can get this merged :/

**Review by Artturin — CHANGES_REQUESTED** [review]

File: pkgs/by-name/mf/mfcj880dw-cupswrappe/package.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    sed -i -e '26,306d' $TARGETFOLDER/cupswrappermfcj880dw
+    substituteInPlace $TARGETFOLDER/cupswrappermfcj880dw \
+      --replace "\$ppd_file_name" "$TARGETFOLDER/brother_mfcj880dw_printer_en.ppd"
+
+    CPUSFILTERFOLDER=$out/lib/cups/filter
+    mkdir -p $TARGETFOLDER $CPUSFILTERFOLDER
+    ln -s ${mfcj880dwlpr}/lib/cups/filter/brother_lpdwrapper_mfcj880dw $out/lib/cups/filter/brother_lpdwrapper_mfcj880dw
+    '';
+
+  cleanPhase = ''
```
Comment (Artturin) [inline pkgs/by-name/mf/mfcj880dw-cupswrappe/package.nix:36]: There is no `cleanPhase`

Please apply the changes from https://github.com/NixOS/nixpkgs/pull/291609

**Comment by 6543** [comment]
I hope https://github.com/NixOS/nixpkgs/compare/68ac64303a89e0fafcc25481df50c8e469ed97dc..8f0ce83807107efd3d483b816efa9b4336833556 was the right changes :)

**Review by Artturin — APPROVED** [review]

**Comment by 6543** [comment]
@Artturin thanks for the guidance :+1: 

can/should i backport it to the current stable branch?

**Comment by Artturin** [comment]
> @Artturin thanks for the guidance 👍
> 
> can/should i backport it to the current stable branch?

New packages can be backported https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#changes-acceptable-for-releases

I added the label for you, in the future once you've received the automatic invite to the nixpkgs team you can add the backport label yourself https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#how-to-backport-pull-requests

**Pull request merged**